tests/build-tools.test.js

```diff
@@ -0,0 +1,106 @@
+const { readFileSync, rmSync, mkdirSync, writeFileSync } = require('fs');
+const { resolve } = require('path');
+const { getData } = require('../scripts/tools/extract-tools-github');
+const { convertTools } = require('../scripts/tools/tools-object');
+const { combineTools } = require('../scripts/tools/combine-tools');
+const { buildTools } = require('../scripts/build-tools');
+const { tagsData, manualTools, mockConvertedData, initialToolsData, mockExtractData } = require('../tests/fixtures/buildToolsData');
+
+jest.mock('../scripts/tools/extract-tools-github');
+jest.mock('../scripts/tools/tools-object');
+jest.mock('../scripts/tools/combine-tools');
+
+describe('buildTools', () => {
+    const testDir = resolve(__dirname, 'test_config');
+    const toolsPath = resolve(testDir, 'tools.json');
+    const tagsPath = resolve(testDir, 'all-tags.json');
+    const automatedToolsPath = resolve(testDir, 'tools-automated.json');
+    const manualToolsPath = resolve(testDir, 'tools-manual.json');
+
+    beforeAll(() => {
+        mkdirSync(testDir, { recursive: true });
+
+        writeFileSync(tagsPath, JSON.stringify(tagsData));
+        writeFileSync(automatedToolsPath, JSON.stringify(mockConvertedData));
+        writeFileSync(manualToolsPath, JSON.stringify(manualTools));
+        writeFileSync(toolsPath, JSON.stringify(initialToolsData));
```

Comment:
Why are you writing the files initially? Aren't they supposed to be created by the script on the go?

You can mock the functions and return the value you placed inside fixtures, and the functions would handle the rest of things.
Remember what we did inside other tested scripts (example build-meetings.test.js)

Create test directory -> mockResolvedValue inside test case -> pass on the file path -> check whether the written file's content is equal to the content defined in fixtures.